### PR TITLE
docs: add documentation about rabbitmq needed resources

### PR DESCRIPTION
**Pull request by melflitty**
Fix: https://github.com/aneoconsulting/ArmoniK/issues/941

### CODE REVIEW

File: .docs/content/1.concepts/8.rabbitmq.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+[Rabbitmq](https://www.rabbitmq.com/) is an open-source message broker, it allows to define queues to which applications connect in order to transfer messages. Rabbitmq originally implements the [Advanced Message Queuing Protocol](https://www.amqp.org/) (AMQP). But it supports other protocols such as [Streaming Text Oriented Messaging Protocol](https://stomp.github.io/) (STOMP) and [MQ Telemetry Transport](https://mqtt.org/) (MQTT). 
+
+[ArmoniK.Core](https://github.com/aneoconsulting/armonik.core) uses rabbitmq message broker to manage task execution. After they are submitted, tasks wait in queues to be executed. 
+
+The protocols supported by rabbitmq in ArmoniK.Core are:
+
+- the original rabbitmq implemented protocol AMQP 0.9.1 
+- AMQP 1.0.0
+
+The configuration of rabbitmq in ArmoniK.Core discussed in this article concern the AMQP 1.0.0 protocol with rabbitmq version 3.11.13.
```
Comment (ddiakiteaneo) [inline .docs/content/1.concepts/8.rabbitmq.md:12]:
```suggestion
The configuration of rabbitmq in ArmoniK.Core discussed in this article concerns the AMQP 1.0.0 protocol with rabbitmq version 3.11.13.
```

File: .docs/content/1.concepts/8.rabbitmq.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+![Alt text](https://blogs.sap.com/wp-content/uploads/2020/10/rabbitmq.png "Rabbitmq architecture")
+
+## Rabbitmq flow control
+
+Sometimes, when a producer (publisher) sends messages to RabbitMQ, it can send them at a faster rate than the consumers can consume them. This can happen if consumers slow down, or after a sudden message spike ingress. If the producer continues to send messages without any kind of flow control, the queue can become congested. In systems with limited resources, this can be critical as it can lead to dropped messages and reduced performance.
+
+Rabbitmq has a [flow control  mechanism](https://blog.rabbitmq.com/posts/2020/05/quorum-queues-and-flow-control-the-concepts/)  to prevent such scenarios. This mechanism is a way to apply back pressure to publishers to avoid overloading consumers. It allows to reduce the speed of the connections when publishing is too quick.
```
Comment (ddiakiteaneo) [inline .docs/content/1.concepts/8.rabbitmq.md:28]: "pressure on publishers"

File: .docs/content/1.concepts/8.rabbitmq.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+This applies also to consumers attempting to retrieve messages from the queue. When a consumer is ready to receive messages, it sends a request for a certain amount of messages called a credit. Each consumer has a credit limit, which determines the maximum number of messages that the consumer can receive at any given time.
+
+The credit system ensures that messages are not lost or dropped due to network congestion or other issues. It allows RabbitMQ to manage the rate at which messages are delivered to consumers, and helps to prevent consumers from becoming overwhelmed with too many messages at once.
+
+
+## Limitations of rabbitmq in ArmoniK
+
+ArmoniK is supposed to handle cases where a large number of tasks are submitted, waiting to be executed, but executing is slow. Therefore, task submission may outpace task execution. And we need to handle that properly. That's why we allow the publishers to outpace consumers. The current configuration of rabbitmq flow control in ArmoniK.Core allow this to happen.
```
Comment (ddiakiteaneo) [inline .docs/content/1.concepts/8.rabbitmq.md:41]: You mean "... execution is slow " ?

File: .docs/content/1.concepts/8.rabbitmq.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+### Configuration of rabbitmq in ArmoniK.Core
+
+In ArmoniK, the consumers are given a credit of 2 messages, which means they can consume two messages at a time from the queue. This configuration was necessary to guarantee the load balancing between consumers. 
+
+If we allow more messages to be sent on the link, they can be all retrieved by a single consumer while others did not consume anything. So, while one consumer is so busy processing all the messages he retrieved from the queue, other consumers are idle, waiting for new messages to be available on the queue. A much more optimal solution is to have each consumer processing a lower number of messages. By setting the number of messages a consumer can consume at a time to 2, each consumer will retrieve two messages at a time, process them and then come back to the queue. If new messages are available, the consumer can consume two and so on.  
+
```
[the remark this answers is not in the document]
Comment (melflitty) [reply .docs/content/1.concepts/8.rabbitmq.md:48]: I prefer beginning with the number of credits in ArmoniK, then presenting why

**Pull request merged**